tests/unit/expire.tcl

```diff
@@ -571,6 +571,24 @@ start_server {tags {"expire"}} {
                 assert_equal [$primary pexpiretime $key] [$replica pexpiretime $key]
             }
         }
+        test {expired keys which is created in writeable replicas should be delete} {
+            $primary flushall
+            $replica config set replica-read-only no
+            $replica config set appendonly yes
+            $replica set foo bar PX 1
+            wait_for_condition 20 100 {
+                [$replica get foo] eq {}
+            } else {
+                fail "can't delete expired the keys which is create in writeable replicas."
+            }
+            $replica config set appendonly no
+            $replica set foo bar PX 1
+            wait_for_condition 20 100 {
+                [$replica get foo] eq {}
+            } else  {
+                fail "can't delete expired the keys which is create in writeable replicas."
+            }
+        }
```

Comment:
i think we can simplify the test to something like this, WDYT?
```suggestion
    test {expired keys which is created in writeable replicas should be delete} {
        $primary flushall
        $replica config set replica-read-only no
        foreach {yes_or_no} {yes no} {
            $replica config set appendonly $yes_or_no
            $replica set foo bar PX 1
            after 1
            assert_equal {} [$replica get foo]
        }
    }
```

Reply:
+1

Reply:
cool, foreach is beautifull ^_^

Reply:
when [$replica get foo] get a key, it should always get a nil, right? regardless of whether the key is actually deleted (as long as the key has been logically expired)

or do you want to test this case: let the active expiry actually delete the key? The GET will never delete the key. in this case, you can take a look with `debug set-active-expire` command

Reply:
> do you want to test this case: let the active expiry actually delete the key?

Yes, my intention is to only test the active expiry, maybe I didn’t realize it myself until your suggestion code:-). And I think it is more appropriate to use `after` than `wait_for_condition ` as you suggest, just set the waiting time to be greater than 100ms